test/e2e/common/expansion.go

```diff
@@ -388,16 +449,11 @@ var _ = framework.KubeDescribe("Variable Expansion", func() {
 
 		By("updating the pod")
 		podClient.Update(podName, func(pod *v1.Pod) {
-			pod.ObjectMeta.Annotations = map[string]string{"mysubpath": "mypath"}
+			pod.ObjectMeta.Annotations = map[string]string{"mysubpath": "foo"}
 		})
 
-		By("waiting for pod running")
-		err = framework.WaitTimeoutForPodRunningInNamespace(f.ClientSet, pod.Name, pod.Namespace, framework.PodStartShortTimeout)
-		Expect(err).NotTo(HaveOccurred(), "while waiting for pod to be running")
-
-		By("deleting the pod gracefully")
-		err = framework.DeletePodWithWait(f, f.ClientSet, pod)
-		Expect(err).NotTo(HaveOccurred(), "failed to delete pod")
+		By("waiting for pod and container restart")
+		waitForPodContainerRestart(f, pod, "/subpath_mount/test.log", "/volume_mount/foo/test.log")
```

Comment:
Let's not confuse two different test cases.

I think this test case should only test Pod bringup from a failed state, and not a container restart.

Reply:
I added the restart to try and remove the failure in the alpha suite - I have reverted that test to its original form. The container restart code was lifted from the subpath suite, so I have changed this version to remove the secondary container and removed the obsolete subpath from the init container